src/bindings/swig/python3/kdb.i

```diff
@@ -38,54 +42,45 @@
 
 // constructors
 %pythonprepend kdb::Key::Key {
-  orig = []
-  if len(args):
-    orig = args[1:]
-    args = [ args[0] ]
+  value = None
+  meta  = {}
+  # check for copy constructor
+  if len(args) and not isinstance(args[0], self.__class__):
+    arg0, args = args[0], args[1:]
+
+    flags = 0
+    args = iter(args)
+    for arg in args:
+      if arg == KEY_END:
+        break
+      elif arg == KEY_SIZE:
+        # ignore value
+        next(args)
+      elif arg == KEY_VALUE:
+        value = next(args)
+      elif arg == KEY_FUNC:
+        #TODO swig directors?
+        raise TypeError("Unsupported meta type")
+      elif arg == KEY_FLAGS:
+        flags = next(args)
+      elif arg == KEY_META:
+        k = next(args)
+        meta[k] = next(args)
+      elif isinstance(arg, int):
+        warnings.warn("Deprecated option in keyNew: {0}".format(arg),
+          RuntimeWarning)
+        flags |= arg
+      elif DEBUG:
```

Comment:
warnings.warn seem to already handle debug/switch off warnings, so no need for

     elif DEBUG:

here